src/main/java/seedu/address/model/commission/Commission.java

```diff
@@ -0,0 +1,139 @@
+package seedu.address.model.commission;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.customer.Customer;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Commission in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Commission {
+
+    // Identity fields
+    private final Title title;
+    private final Description description;
+    private final Fee fee;
+    private final Deadline deadline;
+    private final CompletionStatus completionStatus;
+    private final Set<Tag> tags = new HashSet<>();
+    private Customer customer;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Commission(Title title, Description description, Fee fee, Deadline deadline, Set<Tag> tags,
+                      CompletionStatus completionStatus, Customer customer) {
+        requireAllNonNull(title, description, fee, deadline, tags, completionStatus);
+        this.title = title;
+        this.description = description;
+        this.fee = fee;
+        this.deadline = deadline;
+        this.tags.addAll(tags);
+        this.completionStatus = completionStatus;
+        this.customer = customer;
+    }
+
+    public Title getTitle() {
+        return title;
+    }
+
+    public Description getDescription() {
+        return description;
+    }
+
+    public Fee getFee() {
+        return fee;
+    }
+
+    public Deadline getDeadline() {
+        return deadline;
+    }
+
+    public CompletionStatus getCompletionStatus() {
+        return completionStatus;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    public Customer getCustomer() {
+        return customer;
+    }
+
+    public void setCustomer(Customer customer) {
+        this.customer = customer;
+    }
+
+    /**
+     * Returns true if both commissions have the same title.
+     * This defines a weaker notion of equality between two commissions.
+     */
+    public boolean isSameCommission(Commission otherCommission) {
+        if (otherCommission == this) {
+            return true;
+        }
+
+        return otherCommission != null
+                && otherCommission.getTitle().equals(getTitle());
+    }
+
+    /**
+     * Returns true if both commissions have the same fields.
+     * This defines a stronger notion of equality between two commissions.
+     */
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Should we consider description, completionStatus or customer when checking for equality?